spec/industry_mapping_client_spec.rb

```diff
@@ -17,4 +17,48 @@
     end
   end
 
+  describe '.configure' do
+    describe 'log_failed option' do
+      subject { described_class.instance_variable_get('@log_failed') }
+      let(:base_url) { 'example.com' }
+
+      let(:topic) { 'fake_topic' }
+      let(:source) { 'fake_source' }
+      let(:query) do
+        { query: { topic: topic,
+                   source: source,
+                   log_failed: log_failed } }
+      end
+
+      context 'by default' do
+        it { is_expected.to be_falsey }
+      end
+
+      context 'when log_failed configured' do
+        let(:log_failed) { true }
+
+        before do
+          IndustryMappingClient.configure(log_failed: log_failed)
+          IndustryMappingClient.configure(base_url: base_url)
+          expect(HTTParty).to receive(:get).with(base_url, query).and_return(nil)
+        end
+
+        context 'when set to true' do
+          it { expect(IndustryMappingClient.map_industry(topic, source)).to be_nil }
+        end
+
+        context 'when set to false' do
+          let(:log_failed) { false }
+
+          it { expect(IndustryMappingClient.map_industry(topic, source)).to be_nil }
+        end
+
+        context 'when set invalid value' do
+          let(:log_failed) { nil }
+
+          it { expect(IndustryMappingClient.map_industry(topic, source)).to be_nil }
+        end
+      end
+    end
+  end
```

Comment:
@baraabourghli the point of the specs is to show that the relevant code works as expected. In this case, it's not that easy to really see that failed lookups are being logged, since we're sending requests to IM, an external system. The best we can do is check that `@config[:log_failed]` has been set correctly by the configuration code.

Your specs are doing this though. Mocking HTTParty and calling map_industry doesn't tell us that we're telling IM to log failed lookups. These specs need to tell us that. Follow the example on line 33 and test that in each case, `@config[:log_failed]` is set correctly.